streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java

```diff
@@ -98,6 +104,49 @@ public void shouldThrowProcessorStateExceptionOnOpeningReadOnlyDir() throws IOEx
         subject.openDB(tmpContext);
     }
 
+    @Test
+    public void shouldPutAll() {
+        List<KeyValue<String, String>> entries = new ArrayList<>();
+        entries.add(new KeyValue<String, String>("1", "a"));
+        entries.add(new KeyValue<String, String>("2", "b"));
+        entries.add(new KeyValue<String, String>("3", "c"));
+
+        subject.init(context, subject);
+        subject.putAll(entries);
+        subject.flush();
+
+        assertEquals(subject.get("1"), "a");
+        assertEquals(subject.get("2"), "b");
+        assertEquals(subject.get("3"), "c");
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionOnNullPut() {
```

Comment:
We don't throw NPE anywhere in the `RocksDBStore`, so that means these are coming from `RocksDB` itself. It might be better for us to do `Objects.requireNonNull(...)` in the relevant methods